azure-service-operator.yaml

```diff
@@ -0,0 +1,71 @@
+package:
+  name: azure-service-operator
+  version: 2.12.0
+  epoch: 0
+  description: Azure Service Operator to provision Azure resources and connect your applications to them from within Kubernetes.
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+      - wolfi-baselayout
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/Azure/azure-service-operator
+      tag: v${{package.version}}
+      expected-commit: 5b5d690fd68451d0dcf11896a3122e216c18452d
+
+  - uses: go/bump
+    with:
+      deps: |-
+        golang.org/x/crypto@v0.35.0
+        golang.org/x/net@v0.36.0
+        github.com/golang-jwt/jwt/v4@v4.5.2
+        golang.org/x/oauth2@v0.27.0
+        google.golang.org/protobuf@v1.33.0
+
+  - uses: go/build
+    with:
+      packages: main.go
+      output: ${{package.name}}
+
+update:
+  enabled: true
+  github:
+    identifier: Azure/azure-service-operator
+    strip-prefix: v
+    tag-filter: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - git
+    environment:
+      POD_NAMESPACE: default
+      AZURE_CLIENT_ID: foo
+      AZURE_CLIENT_SECRET: secret
+      AZURE_TENANT_ID: bar
+      AZURE_SUBSCRIPTION_ID: baz
+  pipeline:
+    - name: Smoke test
+      runs: |
+        azure-service-operator --help
+    - uses: test/kwok/cluster
+    - name: Install CRDs
+      runs: |
+        git clone https://github.com/Azure/azure-service-operator
+        cd azure-service-operator
+        git checkout v2.12.0
+        kubectl apply -f charts/azure-service-operator/crds
+    - name: Test operator manager startup
+      uses: test/daemon-check-output
+      with:
+        start: /usr/bin/${{package.name}} --kubeconfig ~/.kube/config
+        timeout: 30
+        expected_output: |
+          Starting Controller
+          Starting workers
+          Stopping and waiting for HTTP servers
```

Comment:
Can we have a `post:` to make simple `wait-for-it` and `curl | grep` thing?

Reply:
The manager will fail just after it, so we don't expect to really come up. It will need access to Azure API.
Do you have something in mind to test further?

Reply:
Gotcha, that makes sense.